README.md

```diff
@@ -1,135 +1,170 @@
 
-```
+To train on a different dataset, simply change the arguments:
+
+- `dataset_name`: Specify a dataset from the Hub on which to fine-tune your model.  
+- `dataset_config_name`: If the dataset contains multiple language subsets, select the language ID of the subset you want to train on.
+
+If the data set doesn't contain a train and a validation split, you can automatically split it into train-val-test 
+(90-10-10) by setting the argument: 
+
+- `split_on_subset`: Specify the subset of the dataset you want to split into train-val-set.
+
+> [!IMPORTANT]
+> For now, this library can only be used to fine-tune the [segmentation model](https://huggingface.co/pyannote/segmentation-3.0) from the [speaker diarization pipeline](https://huggingface.co/pyannote/speaker-diarization-3.1). 
+> Future work will aim to help optimise the hyperparameters of the entire pipeline. 
+
+## Evaluation
+
+The script [`test_segmentation.py`](test_segmentation.py) can be used to evaluate a fine-tuned model on a diarization
+dataset. In the following example, we evaluate the fine-tuned model from the previous step on the test split of the 
+CallHome Japanese dataset:
+
+```bash
 python3 test_segmentation.py \
-      --dataset_name=diarizers-community/callhome" \
-      --dataset_config_name=jpn" \ 
-      --do_split_on_subset=data" \
-      --model_name_or_path=diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn" \ 
-      --preprocessing_num_workers=2  
+    --dataset_name=diarizers-community/callhome \
+    --dataset_config_name=jpn \ 
+    --split_on_subset=data \
+    --test_split_name=test \
+    --model_name_or_path=diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn \
+    --preprocessing_num_workers=2
 ```
 
-## Use in pyannote
+## Inference with pyannote
 
-Use the fine-tuned segmentation model within a speaker diarization pipeline: 
+The fine-tuned segmentation model can easily be loaded into the `pyannote` speaker diarization pipeline for inference. 
+To do so, we load the pre-trained segmentation pipeline, and subsequently override the segmentation model with our 
+fine-tuned checkpoint:
 
 ```python
 from diarizers import SegmentationModel
 from pyannote.audio import Pipeline
+from datasets import load_dataset
+import torch
+
+device = torch.device("cuda:0") if torch.cuda.is_available() else torch.device("cpu")
 
 # load the pre-trained pyannote pipeline
 pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization-3.1")
+pipeline.to(device)
 
 # replace the segmentation model with your fine-tuned one
 model = SegmentationModel().from_pretrained("diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn")
 model = model.to_pyannote_model()
-pipeline.segmentation_model = model
+pipeline.segmentation_model = model.to(device)
+
+# load dataset example
+dataset = load_dataset("diarizers-community/callhome", "jpn", split="data")
+sample = dataset[0]["audio"]
+
+# pre-process inputs
```

Comment:
This example gets a bit complicated trying to format the sample in way expected by pyannote - happy to revert to just calling `diarization_output = pipeline("audio.wav")` if you think it's cleaner